MekHQ/docs/DistributedMekHQ.md

```diff
@@ -0,0 +1,310 @@
+- **Remote Campaign**: any Campaign not running as part of the local MekHQ client.
+- **Timeline**: the latest date among the connected Campaigns.
+- **Teams**: connected Campaigns which cannot be generated as OpFor in a contract.
+- **Trading**: a mechanism by which connected Campaigns can exchange parts, units,
+personnel, and money in exchange for any combination of the same.
+
+### Common Timeline
+The fundamental feature of Distributed MekHQ is a common timeline shared among
+the connected campaigns. Ultimately, each campaign will be managed and operated
+independently, EXCEPT for advancing time. This is accomplished through one simple
+invariant:
+> No campaign may be further along in time than the Host campaign.
+
+This invariant is held through the following means:
+1. Only the Host Campaign may advance the day without restrictions.
+2. Remote Campaigns may advance the day without contacting the Host Campaign if
+   they are at least one day behind the Host Campaign.
+3. Remote Campaigns may inform the Host Campaign they are ready to advance
+   the day, but their local campaign will not advance until their client has
+   been informed that the Host Campaign has moved beyond their local date.
+
+#### Corner Cases
+1. A Remote Campaign connects and is AHEAD of the Host Campaign.
+    a. The Remote Campaign SHALL NOT advance time until the Host Campaign
+       has moved beyond their local date.
+2. A Remote Campaign connects and is markedly behind the Host Campaign.
+    a. The Remote Campaign MAY advance time in bulk, but not past the
+       Host Campaign.
+    b. The Remote Campaign MAY be given the option to "Jump" to the Host
+       Campaign's timeline, skipping all the usual checks. This should
+       only be an option in GM Mode.
+3. The Host Campaign disconnects.
+    a. The Remote Campaign will be given the option to continue playing
+       without a Host Campaign. If the timelines diverge before they
+       reconnect, the invariants will be held through the aforementioned
+       means.
+
+### Common Scenarios and Contracts
+This objective is TBD on its design details.
+
+### Trading
+If you give a mouse a cookie, it is said that they will want a glass of milk.
+Trading is the natural extension to Distributed MekHQ.
+
+Initial support will be limited to the following actions:
+- Same-day wiring of funds.
+- JumpPath distance + 1 day shipping of parts for a fee.
+- JumpPath distance + 1 day shipping of STOCK units for a fee.
+- JumpPath distance + 1 day transit of personnel for a fee.
+
+Subsequent support will be added for the following actions:
+- JumpPath distance + 1 day shipping of CUSTOM units for a fee.
+
+Support may be considered for the following actions:
+- Usage of company JumpShips and DropShips to transport traded items.
+
+### Centralized Markets
+Once Campaigns are connected they should operate under the same markets.
+First-come, first-serve will be used, along with diversifying the selections
```

Comment:
Do we want to expand the number of potential personnel in the market for this? I find Dylan's Method generates lots of people, but I've only seen a few personnel generated from the other variants.

Maybe we should create a new market variant based on one of the methods that is specifically for Distributed MekHQ? Or we could create a variant of each one, to provide options?

Reply:
I imagine we'd pick one and expand it to directly support this. The rest would be "local".